tests/integration/references/has-many-test.js

```diff
@@ -662,6 +662,44 @@ test("load() fetches link when remoteType is link", function(assert) {
   });
 });
 
+test("load() fetches link when remoteType is link but an empty set of records is returned", function(assert) {
+  var done = assert.async();
+
+  env.adapter.findHasMany = function(store, snapshot, link) {
+    assert.equal(link, "/families/1/persons");
+
+    return Ember.RSVP.resolve({ data: [] });
+  };
+
+  var family;
+  run(function() {
+    family = env.store.push({
+      data: {
+        type: 'family',
+        id: 1,
+        relationships: {
+          persons: {
+            links: { related: '/families/1/persons' }
+          }
+        }
+      }
+    });
+  });
+
+  var personsReference = family.hasMany('persons');
+  assert.equal(personsReference.remoteType(), "link");
+
+  run(function() {
+    personsReference.load().then(function(records) {
+      assert.ok(records instanceof DS.ManyArray, "push resolves with the referenced records");
+      assert.equal(get(records, 'length'), 0);
+      assert.equal(get(personsReference.value(), 'length'), 0);
+
+      done();
```

Comment:
lets not use `done`, instead use promises directly. That way when the promise rejects, the test fails.